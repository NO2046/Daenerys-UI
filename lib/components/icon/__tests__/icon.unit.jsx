import React from 'react'
import Icon from '../icon'
import { mount } from 'enzyme'
import * as renderer from 'react-test-renderer'

describe('Icon',() => {
  it('icon render successfully', () => {
    const json = renderer.create(<Icon name="cake"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('icon can be onClick', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="cake" onClick={fn} />)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})


