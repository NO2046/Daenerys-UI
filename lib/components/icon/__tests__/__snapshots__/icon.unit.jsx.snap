// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon icon render successfully 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#cake"
  />
</svg>
`;
